lib/reverse_sentence.rb

```diff
@@ -1,4 +1,50 @@
 # A method to reverse the words in a sentence, in place.
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  string_reverse(my_sentence)
+  reverse_words(my_sentence)
 end
+
+def string_reverse(my_string)
+  return my_string if my_string.nil?
+  return my_string if my_string.length <= 1
+
+  i = 0
+  j = my_string.length - 1
+  b = 0
+  while i < j
+    b = my_string[i]
+    my_string[i] = my_string[j]
+    my_string[j] = b
+    i += 1
+    j -= 1
+  end
+  return my_string
+end
+
+def reverse_words(my_words)
```

Comment:
I've the same comments as the ones I added to https://github.com/Ada-C10/reverse_words/pull/24#issuecomment-439167171

And accordingly, you may want to revisit the algorithm and the complexities for this assignment.